File: test/Test_PlatypusMilk.cpp
#include "../src/PlatypusMilk.cpp"
#include <gtest/gtest.h>

using namespace std;

TEST(PlatypusMilkTest, AllFunctionalityofPlatypusMilk){
    PlatypusMilk testPM(25, "Ferry");
    ASSERT_EQ("Ferry", testPM.getProducer());
    ASSERT_EQ(PLATYPUS_MILK, testPM.getObjectType());
    ASSERT_EQ(FARMPRODUCT, testPM.getProductType());
    ASSERT_EQ(25, testPM.getPrice());
}

File: Java/src/test/java/engisubs/gameobject/product/farmproduct/milk/TestPlatypusMilk.java
package engisubs.gameobject.product.farmproduct.milk;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import engisubs.gameobject.GameObject.GameObjectType;
import engisubs.gameobject.product.Product.ProductType;
public class TestPlatypusMilk{
    private PlatypusMilk test;
    @Before
    public void initObject(){
        test = new PlatypusMilk(25, "Name");
    }

    @Test
    public void instantiationTest(){
        Assert.assertEquals("Name", test.getProducer());
        Assert.assertEquals(GameObjectType.PLATYPUS_MILK, test.getObjectType());
        Assert.assertEquals(ProductType.FARMPRODUCT, test.getProductType());
        Assert.assertEquals(25, test.getPrice());
    }

}